packages/service-provider-core/src/connect-info.ts

```diff
@@ -57,6 +58,7 @@ export default function getConnectInfo(
     is_enterprise: getBuildInfo.isEnterprise(buildInfo),
     auth_type,
     is_data_federation,
+    is_stream: getBuildInfo.isAtlasStream(uri),
```

Comment:
Is there a more reliable check than a URL check? We only do URL checks because there’s nothing we can easily get programmatically out of `buildInfo` etc., but here it seems like we have the opportunity to actually add a proper server-side API for unambiguously indicating a streams processing environment

Reply:
Right now, there isn't. Maybe can potentially change ADF to add something in the response of `buildInfo` to indicate this is a stream instance.

Reply:
Mm that's a good point, people do all sort of things to connect, we do detect atlas from the url only, but should be only for telemetry or presentation purposes, if a feature depends on it we may want something returned from the server.

Reply:
I definitely agree returning something from server is more reliable and better. But at the moment, this is only used to change the default prompt which user can change. And `sp` is always there regardless of whether this is actually a stream instance. Which makes me wondering how important it is to get it 100% correct.